pallets/pips/src/lib.rs

```diff
@@ -502,10 +502,25 @@ decl_module! {
         /// changes the network in someway. A minimum deposit is required to open a new proposal.
         ///
         /// # Arguments
-        /// * `proposal` a dispatchable call
-        /// * `deposit` minimum deposit value
-        /// * `url` a link to a website for proposal discussion
-        #[weight = SimpleDispatchInfo::FixedNormal(5_000_000)]
+        /// * `proposal` - a dispatchable call
+        /// * `deposit` - minimum deposit value
+        /// * `url` - a link to a website for proposal discussion
+        ///
+        /// # Weight
+        /// `5_000_000 + 100 * proposal.len()`
+        #[weight = FunctionOf(
+            |(proposal, _, _, _, _): (
+                &Box<T::Proposal>,
+                &BalanceOf<T>,
+                &Option<Url>,
+                &Option<PipDescription>,
+                &Vec<Beneficiary<T::Balance>>
+            )| {
+                5_000_000 + 100 * u32::try_from(proposal.encode().len()).unwrap_or_default()
```

Comment:
@vkandy Have you tried with `std::mem::size_of_val(*proposal)`? 
If that works, we could eliminate the `encode` part.